src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -1396,6 +1396,14 @@ void ImportFrontendModelArray(const void *onnxBuffer, int size,
 
 void ImportFrontendModelFile(std::string model_fname, MLIRContext &context,
     OwningModuleRef &module, std::string *errorMessage, ImportOptions options) {
+  // try to open the input file to check if it can be opened in advance.
+  std::fstream test;
+  test.open(model_fname);
+  if (!test) {
```

Comment:
I think this should be:
```
if (!test.is_open()) {
```

Thinking more about this ... @negiyas it would be nice if we could avoid opening the file twice. Can you try to see if this would work:

```
  std::fstream input(model_fname, std::ios::in | std::ios::binary);
  if (!input.is_open()) {
    *errorMessage = "Unable to open or access " + model_fname;
    return;
  }
  ```

Reply:
@etiotto It works. Thank you very much for useful comments!

Reply:
@negiyas you are very welcome! Thanks for the fix.